device_backends/DummyBackend/src/DummyBackend.cc

```diff
@@ -144,6 +176,33 @@ namespace ChimeraTK {
     return returnList;
   }
 
+  std::list<boost::function<void(void)>> DummyBackend::findReadCallbackFunctionsForAddressRange(AddressRange addressRange) {
+    // as callback functions are not sortable, we want to loop the multimap only
+    // once.
+    // FIXME: If the same function is registered more than one, it may be executed
```

Comment:
This can be fixed by changing the `std::list` into an `std::set`

Reply:
Are we sure that we want to change this behaviour ? I do not fully understand what this code is doing and what it suppose to do. If the same function is registered with two different address ranges, then maybe we want to call it twice. This is change of behavior so I would prefer to understand it better before just implementing change according to FIXME comment from Jan Timm.

Reply:
Well, this code is introduced by this pull request. It cannot be used anywhere else yet.

Reply:
Yes, you are right. It seems to be kind of logic copy of `findCallbackFunctionsForAddressRange`, but for read operation (not write). It looks that Jan Timm was preparing DummyBackend for something, but it was probably never finished.  Its also why the method setReadCallback was added to DummyRegisterAccessor . 
Code which uses the readCallback part is used only in testDummyBackend which tests the new codeitself, but it seems not to be used in other places in device access. Question is whether this functionality is needed and we should keep it ?

Generic comment is also that the same FIXME comments is in `findCallbackFunctionsForAddressRange` which already is in the master.

There is also now a little naming incosistency. The callbacks for write are stored in std::multimap _writeCallbackFunctions, but functions which operates on this object are just findCallbackFunctionsForAddressRange. Maybe it should be named then: findWriteCallbackFunctionsForAddressRange.

Reply:
Read callback functionality has been removed from this branch. Feature is now in separated branch ([wip/9740-ExtendDummyBackendToSupportReadCallbacks](https://github.com/ChimeraTK/DeviceAccess/tree/wip/9740-ExtendDummyBackendToSupportReadCallbacks)) so I think we can resolve this part of the pull request.